tests/unit/test_operator.py

```diff
@@ -77,10 +94,165 @@ def test_with_relation(harness):
         "app",
         {"_supported_versions": "- v1", "data": yaml.dump(data)},
     )
-    harness.begin_with_initial_hooks()
+    yield harness
+    harness.cleanup()
+
+
+class TestCharm:
+    @patch("charm.KubernetesServicePatch", lambda x, y: None)
```

Comment:
An optional suggestion - there's no right or wrong answer...

This patch is applied to every test function, and you never need to use the `Mocker` it creates.  You can move this `@patch("...")` up to decorate `TestCharm`, and remove it from all the test functions.  
You could almost do the same thing to the `k8s_resource_handler` patch, too, except you occasionally need the Mocker.  Maybe there's a clever way to address that, but I don't know it.

Reply:
Having it on TestCharm may lead into some unexpected behavior when sb will be rewriting it (if she forgets about the class scope of the patch). I was thinking a lot about it. This seemed to be most readable thing. But I am open for discussion.

Reply:
Yep fair point.  Thanks!